### ISSUE TITLE
Revisit ActiveScreenViewModel and hook up Active/EditingScreen and DatabaseService

### ISSUE THREAD

**Issue by holtzmak** [holtzmak/Humane-Transport#178]
Blocked by #130.

This task involves adding the rest of the "glue" between database service and the GUI screens

- [x] Revisit the view model and add the stream subscription to DatabaseService for updates to active (incomplete) ATRs
- [x] Revisit the view model and add the provider `notifyListeners()` needed for updates from database service to reach listeners
- [x] Add the function that, when given an edited ATR, will call DatabaseService to save the ATR
- [x] Add the function that, when called, will call DatabaseService to start a new ATR, then return that ATR

_There also exists options for deleting active ATRs, which will be implemented as part of a later task. Saving is the important one now._

If you're unsure about this task, please ask the group chat for pair programming.

**Comment by MummenRider** [comment]
For this task, I'm going to follow a couple tutorials [Andy Julow](https://www.youtube.com/watch?v=dQHQ1_Jen6c&t=1451s&ab_channel=AndyJulow) and [FilledStack](https://www.youtube.com/watch?v=1PhAPWzGaM4&t=745s&ab_channel=FilledStacks)

**Comment by MummenRider** [comment]
My own road map for finishing this task:
- [x] Save new AtrIdentifier to Firestore
- [x] Update existing AtrIdentifier

_*Note: I will add more to this list_

**Comment by holtzmak** [comment]
After revisiting this task in scrum, it appears some work has been done already: that being the stream update in ActiveVM.
Some of Clark's tasks above I think are related to a bug Sana is fixing on the sister task #177 .

So, what's left here is the connection between ATREditingScreen and the view model. We'll want this VM to be static, so the same model is shared between editing and the screen (not necessary on history screen because the display screen doesn't modify anything). Something that comes to mind is:
1. The view model, on creation, should get the existing records and also be hooked to the DBS stream

So, what's left here is:
- [x] Call this VM in AtrEditingScreen on save, save as complete
- [x] Update AtrEditingScreen to call save when the user backs out of it's screen (or hits submit)
- [x] Make ActiveScreen have a grid view, not list view (likely another PR in hand with #177 because both use the same preview card)
- [x] Add tests

**Comment by holtzmak** [comment]
Tests to be done alongside some refactoring in cleanup as part of #162 or #186 .
